lib/jekyll/site_data/file.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module Jekyll
+  module SiteData
+    class File
+      attr_accessor :context
+      attr_reader :content
+
+      def initialize(site, path, content)
+        @site = site
+        @path = path
+        @content = content
```

Comment:
`@content` throughout the Jekyll codebase generally refers to plain text. Perhaps `@data` or `@structured_content` would help indicate that this is generally not a text value, but rather `False`, `Array`, or `Hash`.